Alloy/MTLSharedGraphicsBuffer.swift

```diff
@@ -0,0 +1,150 @@
+import Foundation
+import Accelerate
+import Metal
+
+@available(iOS 12.0, macOS 10.14, *)
+final public class MTLSharedGraphicsBuffer {
+
+    public enum Error: Swift.Error {
+        case initializationFailed
+        case unsupportedPixelFormat
+    }
+
+    public let texture: MTLTexture
+    public let pixelBuffer: CVPixelBuffer
+    public let buffer: MTLBuffer
+    private(set) public var vImageBuffer: vImage_Buffer
```

Comment:
I look at this and understand this is a design flaw. I think we can make Swift's Strings View inspired APIs here, will be much more flexible